src/main.py

```diff
@@ -0,0 +1,102 @@
+from AlertsAggregator import AlertsAggregator
+from OrefAlertsRetrieve import OrefAlertsRetrieveData  
+from fastapi import FastAPI, HTTPException 
+import pandas as pd
+import requests
+from Errors import RetrieveDataException, WrongSettlementException
+from models import AlertsQueryInput
+from datetime import datetime, timedelta
+app = FastAPI()
+
+CSV_FILE_PATH = "C:/Projects/Python/Projects/BestTimeToShower/data/data.csv"
+dataframe = pd.read_csv(CSV_FILE_PATH)
+getdata = OrefAlertsRetrieveData()
+alert_aggregator = AlertsAggregator(dataframe)
+
+
+
+@app.post("/orefalerts")
+async def get_oref_alert(from_date: str, to_date: str):
+    try:
+        global alert_aggregator
+        getdata.get_oref_alarms(from_date, to_date)
+        df = pd.read_csv(CSV_FILE_PATH)
+        alert_aggregator = AlertsAggregator(df)
+        return {"message": "Up-date the alerts data csv"}
+    except RetrieveDataException as e:
+        raise HTTPException(status_code=404, detail=str(e))
+    except PermissionError as ex:
+        raise HTTPException(status_code=401, detail=str(ex))
+    except Exception as exc:
+        raise HTTPException(status_code=500, detail=str(exc))
+
+
+@app.post("/best_time_to_shower")
+async def find_best_time_to_shower(alerts_demand_obj: AlertsQueryInput):
+    try:
+        best_time = alert_aggregator.best_time_to_shower(alerts_demand_obj.settlement, alerts_demand_obj.start_time, alerts_demand_obj.end_time)
+        return best_time
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+    except WrongSettlementException as settl_err:
```

Comment:
dont shorten names. its bad practice and only juniors do that.

in this case you'd rather use:
we can know by the type that this is WrongSettlementException.

```suggestion
    except WrongSettlementException as ex:
```